### ISSUE TITLE
[BUG] - Select component does not trigger html form change

### ISSUE THREAD

**Issue by micama** [nextui-org/nextui#4111]
### NextUI Version

2.4.8

### Describe the bug

Within a plain html form we have a validation that works through the form's `onChange` event and respective callback method.
The NextUI components like `Input` trigger the `onChange` event of the form, however the `Select` component does not!
This is unexpected. The expectation is for it to behave the same as the native `select` component, i.e. trigger the `onChange` event. 
See the  attached example URL.

Thanks

### Your Example Website or App

https://stackblitz.com/edit/vitejs-vite-vp1lwv?file=src%2FApp.tsx

### Steps to Reproduce the Bug or Issue

1. Use the native components (input, select) => the form `onChange` is triggered and logged
2. Make a selection on nextui Select component => no troggering of the form's `onChange`

### Expected behavior

The NextUI's `Select` component should behave the same as the native one, i.e. trigger it's surrounding form.

### Operating System Version

macOS

### Browser

Chrome

**Comment by responsive-mind** [comment]
Yes, I have the same issue. The Select component not triggering the form's onChange event is unexpected. It should behave like the native select element for consistency with standard HTML behavior

**Comment by micama** [comment]
The same behavious (not triggering the form event) is also observed for the `Checkbox` component.

**Comment by mdfaizanahmed786** [comment]
I can work on this issue. Let me know if you have any questions. Thank you!

**Comment by wingkwong** [comment]
Closing in favor of v3. If the issue is reproducible in v3, please create a new one with a minimal reproducible example.
